server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/SameShardAllocationDecider.java

```diff
@@ -91,23 +91,12 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
                     continue;
                 }
                 // check if its on the same host as the one we want to allocate to
-                boolean checkNodeOnSameHostName = false;
-                boolean checkNodeOnSameHostAddress = false;
-                if (Strings.hasLength(checkNode.node().getHostAddress()) && Strings.hasLength(node.node().getHostAddress())) {
-                    if (checkNode.node().getHostAddress().equals(node.node().getHostAddress())) {
-                        checkNodeOnSameHostAddress = true;
-                    }
-                } else if (Strings.hasLength(checkNode.node().getHostName()) && Strings.hasLength(node.node().getHostName())) {
-                    if (checkNode.node().getHostName().equals(node.node().getHostName())) {
-                        checkNodeOnSameHostName = true;
-                    }
-                }
-                if (checkNodeOnSameHostAddress || checkNodeOnSameHostName) {
+                assert Strings.hasLength(checkNode.node().getHostAddress()) : checkNode;
+                assert Strings.hasLength(node.node().getHostAddress()) : node;
```

Comment:
NIT: `node.node().getHostAddress()` might be asserted once before the loop

Reply:
Good point, fixed in d3cbade5ae04939ad89d3d2104e0f00393db1df2.

Not really sure these asserts are useful outside the context of this change. I imagine we'll see them in a year or two and wonder why on earth they're there...